### DISCUSSION TITLE
Auth during build

### DISCUSSION THREAD

**Discussion by beisenherz** [ExpediaGroup/graphql-kotlin#1540] [General]
Is there a way to use a oauth login for the build process? I need to switch the endpoint on which the graphql endpoint is exposed, but the new one is only available with OAuth Validation. Or would that be a use case for the schema files?
Is there a kind of 'best practice' to keep the schema file up to date, if this is the only way with your client?

Would be glad for any information regarding secured graphQL endpoints :) If you have further questions I'd be happy to answer any questions or assist in a potential feature :)

**Comment by dariuszkuc** [comment]
Hello 👋 
You can pass headers to `graphqlDownloadSDL`/`graphqlDownloadSDL` tasks (or their equivalent Maven goals) so you could definitely include auth header (that was actually the primary use case for supporting headers) but performing full OAuth handshake is outside of the scope of the `graphql-kotlin` plugins. I'm guessing the simplest way to integrate would be to pass the bearer token on the command line and configure your build scripts to read it from the system/env properties.

Technically, you could also write a Gradle task to obtain the bearer token but that might be an overkill (there might be even a community plugin that does it for you).

**Reply by beisenherz** [reply]
Not tried yet:
The easiest solution seems to be something in [this stackoverflow thread](https://stackoverflow.com/questions/24058926/gradle-script-to-call-a-rest-web-service-without-any-3rd-party-plugins-any-poin)

Also there doesn't seem to be a gradle plugin, that does allow using oauth by itself... seem to be missing entirely :/

Passing it via pipeline seems a little controversial, since you'd be able to read a authenticated Bearer token, but that might be our least painful concern. I'll give it a try tomorrow with the stack overflow solution I linked earlier, if it works I'll share the task here for anyone to use later :)

Thank you for your answer :)
